File: AndroidAssignments8/app/src/main/java/com/android/mad/assignments/SecondActivity.java
package com.android.mad.assignments;

import android.os.Bundle;

import androidx.activity.EdgeToEdge;
import androidx.appcompat.app.AppCompatActivity;
import android.widget.TextView;
import androidx.core.graphics.Insets;
import androidx.core.view.ViewCompat;
import androidx.core.view.WindowInsetsCompat;

public class SecondActivity extends AppCompatActivity {

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        EdgeToEdge.enable(this);
        setContentView(R.layout.second_activity);

        // Получаем переданные данные
        String transmittedString = getIntent().getStringExtra(TRANSMITTED_STRING);
        int transmittedInt = getIntent().getIntExtra(TRANSMITTED_INT, -1);           // -1 — значение по умолчанию
        boolean transmittedBoolean = getIntent().getBooleanExtra(TRANSMITTED_BOOLEAN, false);

        TextView textView = findViewById(R.id.second_activity_text_view);

        textView.setText(
                "These values were passed from previous screen\n\n" +
                        "transmittedString: " + transmittedString + "\n" +
                        "transmittedInt: " + transmittedInt + "\n" +
                        "transmittedBoolean: " + transmittedBoolean
        );
    }

    public static final String TRANSMITTED_STRING = "transmittedString";
    public static final String TRANSMITTED_INT    = "transmittedInt";
    public static final String TRANSMITTED_BOOLEAN = "transmittedBoolean";
}

File: MovieTracker_Lab6_src/app/src/main/java/com/example/movietracker/SecondActivity.kt
package com.example.movietracker

import android.content.Intent
import android.os.Bundle
import android.util.Log
import android.widget.Button
import android.widget.EditText
import android.widget.TextView
import android.widget.Toast
import androidx.appcompat.app.AppCompatActivity

class SecondActivity : AppCompatActivity() {

    companion object {
        private const val TAG = "SecondActivity"
    }

    private lateinit var textViewTitle: TextView
    private lateinit var textViewGenre: TextView
    private lateinit var textViewId: TextView
    private lateinit var editTextReview: EditText
    private lateinit var buttonSendReview: Button
    private lateinit var buttonBack: Button

    private var movieTitle = ""

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        Log.d(TAG, "onCreate called")
        setContentView(R.layout.activity_second)

        textViewTitle    = findViewById(R.id.textViewTitle)
        textViewGenre    = findViewById(R.id.textViewGenre)
        textViewId       = findViewById(R.id.textViewId)
        editTextReview   = findViewById(R.id.editTextReview)
        buttonSendReview = findViewById(R.id.buttonSaveReview)
        buttonBack       = findViewById(R.id.buttonBack)

        // Шаг 2: получаем данные из явного Intent
        movieTitle       = intent.getStringExtra(MainActivity.EXTRA_MOVIE_TITLE) ?: "Без названия"
        val genre        = intent.getStringExtra(MainActivity.EXTRA_MOVIE_GENRE) ?: "Жанр не указан"
        val id           = intent.getLongExtra(MainActivity.EXTRA_MOVIE_ID, 0L)

        textViewTitle.text = "Фильм: $movieTitle"
        textViewGenre.text = "Жанр: $genre"
        textViewId.text    = "ID: $id"
        Log.d(TAG, "Received: title=$movieTitle, genre=$genre, id=$id")

        savedInstanceState?.let {
            editTextReview.setText(it.getString("saved_review", ""))
        }

        // Шаг 3: кнопка "Сохранить рецензию" — возвращает данные через setResult
        buttonSendReview.setOnClickListener {
            val review = editTextReview.text.toString().trim()
            if (review.isEmpty()) {
                Toast.makeText(this, "Введите рецензию для отправки", Toast.LENGTH_SHORT).show()
                return@setOnClickListener
            }
            // Формируем Intent с результатом и отправляем обратно в MainActivity
            val resultIntent = Intent().apply {
                putExtra(MainActivity.EXTRA_RESULT_TITLE,  movieTitle)
                putExtra(MainActivity.EXTRA_RESULT_REVIEW, review)
            }
            setResult(RESULT_OK, resultIntent)
            Log.d(TAG, "setResult RESULT_OK: review=$review")
            finish()  // закрываем SecondActivity, результат уходит в MainActivity
        }

        // Кнопка "Назад" — закрываем без результата
        buttonBack.setOnClickListener {
            setResult(RESULT_CANCELED)
            Log.d(TAG, "RESULT_CANCELED, finishing")
            finish()
        }
    }

    override fun onStart()   { super.onStart();   Log.d(TAG, "onStart called")   }
    override fun onResume()  { super.onResume();  Log.d(TAG, "onResume called")  }
    override fun onPause()   { super.onPause();   Log.d(TAG, "onPause called")   }
    override fun onStop()    { super.onStop();    Log.d(TAG, "onStop called")    }
    override fun onDestroy() { super.onDestroy(); Log.d(TAG, "onDestroy called") }

    override fun onSaveInstanceState(outState: Bundle) {
        super.onSaveInstanceState(outState)
        outState.putString("saved_review", editTextReview.text.toString())
        Log.d(TAG, "onSaveInstanceState: review saved")
    }

    override fun onRestoreInstanceState(savedInstanceState: Bundle) {
        super.onRestoreInstanceState(savedInstanceState)
        editTextReview.setText(savedInstanceState.getString("saved_review", ""))
        Log.d(TAG, "onRestoreInstanceState: review restored")
    }
}
